modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/DDMTemplateBrowserSnifferUpgradeProcessTest.java

```diff
@@ -0,0 +1,164 @@
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+@RunWith(Arquillian.class)
+public class DDMTemplateBrowserSnifferUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		_addDDMTemplateContent();
+	}
+
+	@Test
+	public void testUpgradeDDMTemplateRemoveBrowserSniffer() throws Exception {
+		List<DDMTemplate> ddmTemplateList = _addDDMTemplateContent();
+
+		for (DDMTemplate ddmTemplateIndex : ddmTemplateList) {
+			long templateId = ddmTemplateList.get(
+				(int)ddmTemplateIndex.getTemplateId()
+			).getTemplateId();
+
+			_runUpgrade();
+
+			String scriptResult = _ddmTemplateService.getTemplate(
+				templateId
+			).getScript();
+
+			Assert.assertEquals(
+				_read("updated-ddm-template-browser-sniffer-content.ftl"),
+				scriptResult);
+		}
+	}
+
+	private List<DDMTemplate> _addDDMTemplateContent() throws Exception {
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext();
+
+		long classPK = ParamUtil.getLong(serviceContext, "classPK");
+
+		_ddmTemplateService.addTemplate(
+			_group.getGroupId(), _portal.getClassNameId(DDMTemplate.class),
+			classPK, _portal.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Name"
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Description"
+			).build(),
+			DDMTemplateConstants.TEMPLATE_TYPE_DISPLAY, StringPool.BLANK,
+			TemplateConstants.LANG_TYPE_FTL,
+			_read("ddm-template-browser-sniffer-content.ftl"),
+			serviceContext);
+
+		return _ddmTemplateService.getTemplates(
+			_group.getCompanyId(), _group.getClassNameId(),
+			_portal.getClassNameId(DDMTemplate.class),
+			_portal.getClassNameId(JournalArticle.class), 1);
```

Comment:
@alberttgomes, why do you need to run the method `getTemplates` if the method `addTemplate` already returns the added template?